libraries/script-engine/src/ScriptEngine.cpp

```diff
@@ -117,12 +118,53 @@ static QScriptValue debugPrint(QScriptContext* context, QScriptEngine* engine) {
         message += context->argument(i).toString();
     }
 
-    if (ScriptEngine *scriptEngine = qobject_cast<ScriptEngine*>(engine)) {
+    // was this generated by a script engine? If we don't recognize it then send the message and exit
+    ScriptEngine* scriptEngine = qobject_cast<ScriptEngine*>(engine);
+    if (scriptEngine == nullptr) {
```

Comment:
!scriptEngine (similarly, for some checks below, the coding standard says to prefer not writing out nullptr)